_posts/2022-08-05-hacking-for-cruise-control-ui.md

```diff
@@ -0,0 +1,241 @@
+
+```bash
+# Dockerfile
+FROM nginx
+
+ENV VERSION=0.4.0
+ENV NGINX_HOME=/usr/share/nginx/html/
+
+RUN curl -LO https://github.com/linkedin/cruise-control-ui/releases/download/v${VERSION}/cruise-control-ui-${VERSION}.tar.gz; \
+    tar xvfz cruise-control-ui-${VERSION}.tar.gz -C ${NGINX_HOME} --strip-components=2; \
+    rm -f cruise-control-ui-${VERSION}.tar.gz*; \
+    echo "dev,dev,/kafkacruisecontrol/" > "${NGINX_HOME}"static/config.csv;
+
+EXPOSE 80
+```
+
+Then building and pushing the image to a container registry
+```
+# When building, tag the image with the name of a container registry
+# that is accessible by the Kubernetes cluster.
+docker build . -t <registry>/<cruise-control-ui-image-name>
+
+# Push the image to that container registry
+docker push <registry>/<cruise-control-ui-image-name>
+```
+
+### Deploying the Cruise Control UI pod
+
+Then we can create a Deployment, ConfigMap, and Service for running the Cruise Control UI:
+
+```yaml
+# cruise-control-ui.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: cruise-control-ui
+  labels:
+    app: cruise-control-ui
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: cruise-control-ui
+  template:
+    metadata:
+      labels:
+        app: cruise-control-ui
+    spec:
+      containers:
+      - name: cruise-control-ui
+        image: <registry>/<cruise-control-ui-image-name>
+        ports:
+        - containerPort: 80
+        volumeMounts:
+        - name: cruise-control-ui
+          mountPath: /etc/nginx/conf.d
+      volumes:
+        - name: cruise-control-ui
+          configMap:
+            name: cruise-control-ui
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cruise-control-ui
+  labels:
+    app: cruise-control-ui
+data:
+  default.conf: |2+
+        server {
+        listen       80;
+        listen  [::]:80;
+        server_name  localhost;
+
+        location / {
+            root   /usr/share/nginx/html;
+            index  index.html index.htm;
+        }
+
+        location /kafkacruisecontrol/ {
+            proxy_set_header Host $host;
+            proxy_set_header X-Real-IP $remote_addr;
+            proxy_buffering off;
+            proxy_pass http://my-cluster-cruise-control:9090/kafkacruisecontrol/;
+        }
+        
+        error_page   500 502 503 504  /50x.html;
+        location = /50x.html {
+            root   /usr/share/nginx/html;
+        }
+        }
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: cruise-control-ui
+  labels:
+    app: cruise-control-ui
+spec:
+  selector:
+    app: cruise-control-ui
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 80
+```
+
+Then run it:
+```
+kubectl apply -f cruise-control-ui.yaml -n <namespace>
+```
+
+## Using Cruise Control Front End
+
+After being deployed, port-forward the Cruise Control UI service to access it locally:
+
+```
+kubectl port-forward svc/cruise-control-ui 8090:80
```

Comment:
I guess you are using port-forwarding because of using minikube for your own tests but I don't see you mentioned using minikube across the blog post. You should be clear, otherwise people can use different `Service` and getting a public accessible URL (i.e. they try with AKS, ...)

Reply:
Added a line about minikube